client/ayon_flame/api/plugin.py

```diff
@@ -612,29 +621,57 @@ def _convert_to_marker_data(self):
 
         if not hero_track and self.vertical_sync:
             # driving layer is set as negative match
-            for (_in, _out), hero_data in self.vertical_clip_match.items():
-                """
-                Since only one instance of hero clip is expected in
-                `self.vertical_clip_match`, this will loop only once
-                until none hero clip will be matched with hero clip.
+            for (hero_in, hero_out), hero_data in self.vertical_clip_match.items():  # noqa
+                """ Iterate over all clips in vertical sync match
 
-                `tag_hierarchy_data` will be set only once for every
-                clip which is not hero clip.
+                If clip frame range is outside of hero clip frame range
+                then skip this clip and do not add to hierarchical shared
+                metadata to them.
                 """
-                _hero_data = deepcopy(hero_data)
-                _hero_data.update({"heroTrack": False})
-                if _in <= self.clip_in and _out >= self.clip_out:
-                    data_product_name = hero_data["productName"]
-                    # add track index in case duplicity of names in hero data
-                    if self.product_name in data_product_name:
-                        _hero_data["productName"] = self.product_name + str(
-                            self.track_index)
-                    # in case track name and product name is the same then add
-                    if self.base_product_name == self.track_name:
-                        _hero_data["productName"] = self.product_name
-                    # assign data to return hierarchy data to tag
-                    tag_hierarchy_data = _hero_data
-                    break
+                if self.clip_in < hero_in and self.clip_out > hero_out:
```

Comment:
Shouldn't this be or condition?
```suggestion
                if self.clip_in < hero_in or self.clip_out > hero_out:
```

Reply:
already fixed to `or`